---
title: "quick_clustering"
author: "EPermina"
date: "10 April 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}


if(!require(tidyverse)){
  install.packages("tidyverse")
}

if(!require(SNPRelate)){ 
source("https://bioconductor.org/biocLite.R")
biocLite("SNPRelate")
}

set.seed(1680) # for reproducibility

if(!require(ISLR)) {
  install.packages("ISLR")
}

if(!require(cluster)) {
  install.packages("cluster")
}

if(!require(Rtsne)) {
  install.packages("Rtsne")
}
library(ISLR) # for college dataset
library(cluster) # for gower similarity and pam
library(Rtsne) # for t-SNE plot
library(tidyverse)
library(SNPRelate)
library(reshape2)
```


first_analysis/populations.snps.vcf.filtered.for.non-presence.presumed.0.samplesOnly.txt is a modified populations.snps.vcf file

1. all samples with no coverage of SNPs (./. only columns) were removed
1. vcf header has also been removed
1. 

Finding relevant files on the users computer and reading these in
```{r dynamic reading in files }
CodePath<-getwd()
filePathListOfDirs<-strsplit(CodePath, split="/") %>% as.list()
RootDirPath<-filePathListOfDirs[2:((lengths(filePathListOfDirs))-1)]
filePath <- paste0(getwd(),"/uploads") 
```
Reading data in:
```{r reading the data in}

snps_raw_headless<-read.delim("populations.snps.vcf.txt", sep = "\t", skip = 14)
samples<-grepl("single", names(snps_raw_headless))





df <-SNPsimpleMatrix
 
genotype_sums<- df %>% 
   replace(is.na(.), 0) %>%
   summarise_each(funs(sum))



```


Data manipulation:
read in original vcf file, get read of header
replace ./. (Z in this file) with 0/0:0:0,0
bash manipulation of vcf file (split 0/0:0:0,0 format into 3 columns)

```{r, engine = 'bash', eval = FALSE}
#/3_RADseq/scratch
285 populations.snps.vcf.split.info.genotype.coverage.txt
cat populations.snps.vcf.Z.txt | sed 's/.single//g' | tail -n +15 | cut -f10- > populations.snps.vcf.Z.headless.sampleNames.trimmed.txt
```
naming data columns

```{r}
# read in bash-splitted file
genotype_split<-read.delim("../populations.snps.vcf.split.info.genotype.coverage.txt", header = F, sep = '\t')
genotype_raw<-read.delim("../populations.snps.vcf.Z.headless.sampleNames.trimmed.txt", sep = "\t", header = TRUE) 

genotype_names_in_split <- paste(names, ".1.geno", sep="") # genotype names in split
coverage_names_in_split <- paste(names, ".2.cover", sep="") # coverage names in split
pro<- paste(names, ".3.pro", sep="") # coverage names in split

#bind 3 vectors
 z <- as.vector(rbind(genotype_names_in_split,coverage_names_in_split,pro)) 
names(genotype_split)<-z

coverage_cols<-grepl("2.cover", names(genotype_split))
genotype_cols<-grepl("1.geno", names(genotype_split))
coverage<-genotype_split[coverage_cols]
genotype<-genotype_split[genotype_cols]


```

A function to make a simplified matrix out of vcf file (tbw)
```{r , eval=FALSE, echo=FALSE}
df1<-genotype
pattern0<-"0/0"

#df1 %>% mutate_all(., replace(., grep(pattern0, fixed=TRUE), 0)) %>% head()
#write.table(genotype, file="genotype.txt", sep="\t")

```
sub "0/0" with "0" etc in bash
```{r, engine='bash', eval=FALSE}
cat genotype.txt | sed 's#0/1#0.5#g' | sed 's#0/0#0#g' | sed 's#1/1#1#g' > vcf_version.1/genotype.01.txt
```

read substituted data back in


Clustering - 
 - coverage and genotype
```{r}
genotype<- read.delim("genotype.01.txt", sep = "\t", header = TRUE)
t.coverage<-t(as.matrix(coverage))
t.genotype<-t(as.matrix(genotype))

  # df3 = merge(t.coverage, t.genotype, by.x=row.names(t.coverage), by.y=row.names(t.genotype)) #tbw

# Applying Ward Hierarchical Clustering
d   <- dist(t.coverage, method="euclidean")
fit <- hclust(d, method="ward")
plot(fit)

g4 <- cutree(fit, k = c(4))
g8 <- cutree(fit, k = c(8))


gower_dist <- daisy(t.genotype,
                    metric = "gower",
                    type = list(logratio = 3))
gower_mat <- as.matrix(gower_dist)


tsne_obj <- Rtsne(gower_dist, is_distance = TRUE)
pam_fit <- pam(gower_dist, diss = TRUE, k = 3)
tsne_data <- tsne_obj$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit$clustering),
         name = row.names(t.genotype))

ggplot(aes(x = X, y = Y), data = tsne_data) +
  geom_point(aes(color = cluster))

eucl.obj<-Rtsne(d, is_distance = TRUE)
```
Clustering

using one categorical variable (genotype, NAs defaulted to 0, i.e. ./. becomes 0/0, presumed reference)






```{r}
snp_forsequence_raw_file<-read.delim(file="populations.snps.vcf.txt", header = TRUE, sep = "\t", skip = header_rows_number)
samples<-grepl("single", names(snp_forsequence_raw_file))


separate(data, col, into, sep = "[^[:alnum:]]+", remove = TRUE,
  convert = FALSE, extra = "warn", fill = "warn", ...)


snp_random_genotype <- apply(snp_forsequence_raw_file[samples], 1:2, function(x) as.factor(gsub(x, pattern = "0.333", replacement = rbinom(1, 1, 0.5)) )) 

```