---
title: "vcf documentation"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## GitHub Documents

This is an R Markdown format used for publishing markdown documents to GitHub. When you click the **Knit** button all R code chunks are run and a markdown file (.md) suitable for publishing to GitHub is generated.

### libraries used
```{r libraries}
if(!require(dplyr)){
  install.packages("dplyr")
}
if(!require(ggplot2)){
  install.packages("ggplot2")
}
if(!require(SNPRelate)){ 
source("https://bioconductor.org/biocLite.R")
  
biocLite("SNPRelate")
}
library(dplyr)
library(ggplot2)
library(SNPRelate)
```


## vcf manipulation 

files:
```{r vcf header}

header_rows_number<-14
```


```{r population SNPs}
snp_forsequence_raw_file<-read.delim(file="populations.snps.vcf.txt", header = TRUE, sep = "\t", skip = header_rows_number)
snp_raw_file<-read.delim(file="populations.snps.vcf", header = TRUE, sep = "\t", )
snp_presumed_similarity<-read.delim(file="populations.snps.vcf.filtered.for.non-presence.presumed.0.txt", header = TRUE, sep = "\t" ) # ./. subst for 0/0:4,4:0 (min)

```



```{r code random trees echo=FALSE, eval=FALSE}
sample_columns<-c(10:295)
#snp_for_sequence<-snp_raw_file %>% select( contains("sample")) 
samples<-grepl("single", names(snp_forsequence_raw_file))

  snp_random_genotype <- apply(snp_forsequence_raw_file[samples], 1:2, function(x) as.factor(gsub(x, pattern = "0.333", replacement = rbinom(1, 1, 0.5)) ))  
  
samples<-grepl("single", names(snp_random_genotype))
  snp_random_genotype_factor <- apply(snp_random_genotype[samples], 1:2, as.factor)
  
  df[] <- lapply( df, factor) # the "[]" keeps the dataframe structure
 col_names <- names(df)
# do do it for some names in a vector named 'col_names'
df[col_names] <- lapply(df[col_names] , factor)

  

```

## phylogeny

```{r phylogeny, echo=FALSE, eval=FALSE} 
#vcf to GDS

snpgdsVCF2GDS("populations.snps.chr1.vcf", "populations.snps.chr2.gds")

snpgdsSummary("populations.snps.chr1.gds")

genofile <- openfn.gds("populations.snps.chr1.gds")

#dendogram

dissMatrix  <-  snpgdsDiss(genofile , sample.id=NULL, snp.id=NULL, autosome.only=TRUE,remove.monosnp=TRUE, maf=NaN, missing.rate=NaN, num.thread=10, verbose=TRUE)

snpHCluster <-  snpgdsHCluster(dissMatrix$dist, sample.id=NULL, need.mat=TRUE, hang=0.25)

cutTree <- snpgdsCutTree(snpHCluster, z.threshold=15, outlier.n=5, n.perm = 5000, samp.group=NULL,col.outlier="red", col.list=NULL, pch.outlier=4, pch.list=NULL,label.H=FALSE, label.Z=TRUE, verbose=TRUE)

#pca

sample.id <- read.gdsn(index.gdsn(genofile, "sample.id"))

pop_code <- read.gdsn(index.gdsn(genofile, "sample.id"))

pca <- snpgdsPCA(genofile)

tab <- data.frame(sample.id = pca$sample.id, pop = factor(pop_code)[match(pca$sample.id, sample.id)], EV1 = pca$eigenvect[,1],EV2 = pca$eigenvect[,2],stringsAsFactors = FALSE)

plot(tab$EV2, tab$EV1, col=as.integer(tab$pop),xlab="eigenvector 2", ylab="eigenvector 1")
legend("topleft", legend=levels(tab$pop), pch="o", col=1:nlevels(tab$pop))
```

```{r engine = 'bash', echo=FALSE, eval=FALSE}
Исходные файлы с ридами лежат вот тут 
/home/artem/Osinov/reads/SOMM238_R1_CGATTG.fastq
/home/artem/Osinov/reads/SOMM238_R1_CGGAAT.fastq
/home/artem/Osinov/reads/SOMM238_R1_GAGGTG.fastq
/home/artem/Osinov/reads/SOMM238_R1_TTTTTG.fastq
/home/artem/Osinov/reads/SOMM238_R3_CGATTG.fastq
/home/artem/Osinov/reads/SOMM238_R3_CGGAAT.fastq
/home/artem/Osinov/reads/SOMM238_R3_GAGGTG.fastq
/home/artem/Osinov/reads/SOMM238_R3_TTTTTG.fastq

А по баркодам я разбивал с помощью скрипта /home/artem/Osinov/reads/run_BestRadSplit.sh
Он внутри себя использует скрипт /home/artem/Osinov/reads/BarcodeSplitListBestRadPairedEnd.pl
```



## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
