\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={vcf\_clustering},
            pdfauthor={EPermina},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{{#1}}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{{#1}}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{{#1}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{{#1}}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{{#1}}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{{#1}}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{{#1}}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{{#1}}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{{#1}}}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{{#1}}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{{#1}}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{{#1}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{{#1}}}}
\newcommand{\NormalTok}[1]{{#1}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}
  \title{vcf\_clustering}
  \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
  \author{EPermina}
  \preauthor{\centering\large\emph}
  \postauthor{\par}
  \predate{\centering\large\emph}
  \postdate{\par}
  \date{15 April 2018}


\begin{document}
\maketitle

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{if(!}\KeywordTok{require}\NormalTok{(tidyverse))\{}
  \KeywordTok{install.packages}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: tidyverse
\end{verbatim}

\begin{verbatim}
## -- Attaching packages ---------------------------------- tidyverse 1.2.1 --
\end{verbatim}

\begin{verbatim}
## √ ggplot2 2.2.1     √ purrr   0.2.4
## √ tibble  1.4.2     √ dplyr   0.7.4
## √ tidyr   0.8.0     √ stringr 1.3.0
## √ readr   1.1.1     √ forcats 0.3.0
\end{verbatim}

\begin{verbatim}
## -- Conflicts ------------------------------------- tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{if(!}\KeywordTok{require}\NormalTok{(SNPRelate))\{ }
\KeywordTok{source}\NormalTok{(}\StringTok{"https://bioconductor.org/biocLite.R"}\NormalTok{)}
\KeywordTok{biocLite}\NormalTok{(}\StringTok{"SNPRelate"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: SNPRelate
\end{verbatim}

\begin{verbatim}
## Loading required package: gdsfmt
\end{verbatim}

\begin{verbatim}
## SNPRelate -- supported by Streaming SIMD Extensions 2 (SSE2)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{set.seed}\NormalTok{(}\DecValTok{1680}\NormalTok{) }\CommentTok{# for reproducibility}

\NormalTok{if(!}\KeywordTok{require}\NormalTok{(ISLR)) \{}
  \KeywordTok{install.packages}\NormalTok{(}\StringTok{"ISLR"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: ISLR
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{if(!}\KeywordTok{require}\NormalTok{(cluster)) \{}
  \KeywordTok{install.packages}\NormalTok{(}\StringTok{"cluster"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: cluster
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{if(!}\KeywordTok{require}\NormalTok{(Rtsne)) \{}
  \KeywordTok{install.packages}\NormalTok{(}\StringTok{"Rtsne"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: Rtsne
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(cluster) }\CommentTok{# for gower similarity and pam}
\KeywordTok{library}\NormalTok{(Rtsne) }\CommentTok{# for t-SNE plot}
\KeywordTok{library}\NormalTok{(tidyverse)}
\KeywordTok{library}\NormalTok{(SNPRelate)}
\KeywordTok{library}\NormalTok{(reshape2)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'reshape2'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:tidyr':
## 
##     smiths
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{set.seed}\NormalTok{(}\DecValTok{1680}\NormalTok{) }\CommentTok{# for reproducibility}
\end{Highlighting}
\end{Shaded}

\subsubsection{Data manipulation, reading
in}\label{data-manipulation-reading-in}

first\_analysis/populations.snps.vcf.filtered.for.non-presence.presumed.0.samplesOnly.txt
is a modified populations.snps.vcf file

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  all samples with no coverage of SNPs (./. only columns) were removed
\item
  vcf header has also been removed
\item
  ./. is replaced with 0/0:0:0,0
\end{enumerate}

Finding relevant files on the users computer and reading these in files:
populations.snps.vcf.Z.txt
populations.snps.vcf.Z.headless.sampleNames.trimmed.txt
populations.snps.vcf.split.info.genotype.coverage.txt

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{cat} \NormalTok{populations.snps.vcf.Z.txt }\KeywordTok{|} \KeywordTok{sed} \StringTok{'s/.single//g'} \KeywordTok{|} \KeywordTok{tail} \NormalTok{-n +15 }\KeywordTok{|} \KeywordTok{cut} \NormalTok{-f10- }\KeywordTok{>} \NormalTok{populations.snps.vcf.Z.headless.sampleNames.trimmed.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]

\KeywordTok{cat} \NormalTok{genotype.txt }\KeywordTok{|} \KeywordTok{sed} \StringTok{'s#0/1#0.5#g'} \KeywordTok{|} \KeywordTok{sed} \StringTok{'s#0/0#0#g'} \KeywordTok{|} \KeywordTok{sed} \StringTok{'s#1/1#1#g'} \KeywordTok{>} \NormalTok{genotype.01.txt}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# read in bash-splitted file}
\NormalTok{genotype_split<-}\KeywordTok{read.delim}\NormalTok{(}\StringTok{"../populations.snps.vcf.split.info.genotype.coverage.txt"}\NormalTok{, }\DataTypeTok{header =} \NormalTok{F, }\DataTypeTok{sep =} \StringTok{'}\CharTok{\textbackslash{}t}\StringTok{'}\NormalTok{)}
\NormalTok{genotype_raw<-}\KeywordTok{read.delim}\NormalTok{(}\StringTok{"../populations.snps.vcf.Z.headless.sampleNames.trimmed.txt"}\NormalTok{, }\DataTypeTok{sep =} \StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\DataTypeTok{header =} \OtherTok{TRUE}\NormalTok{) }

\NormalTok{genotype_names_in_split <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(names, }\StringTok{".1.geno"}\NormalTok{, }\DataTypeTok{sep=}\StringTok{""}\NormalTok{) }\CommentTok{# genotype names in split}
\NormalTok{coverage_names_in_split <-}\StringTok{ }\KeywordTok{paste}\NormalTok{(names, }\StringTok{".2.cover"}\NormalTok{, }\DataTypeTok{sep=}\StringTok{""}\NormalTok{) }\CommentTok{# coverage names in split}
\NormalTok{pro<-}\StringTok{ }\KeywordTok{paste}\NormalTok{(names, }\StringTok{".3.pro"}\NormalTok{, }\DataTypeTok{sep=}\StringTok{""}\NormalTok{) }\CommentTok{# coverage names in split}

\CommentTok{#bind 3 vectors}
 \NormalTok{z <-}\StringTok{ }\KeywordTok{as.vector}\NormalTok{(}\KeywordTok{rbind}\NormalTok{(genotype_names_in_split,coverage_names_in_split,pro)) }
\KeywordTok{names}\NormalTok{(genotype_split)<-z}
\end{Highlighting}
\end{Shaded}

Clustering - coverage and genotype

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genotype_split<-}\KeywordTok{read.delim}\NormalTok{(}\StringTok{"genotype_split_object.tab"}\NormalTok{)}
\NormalTok{coverage_cols<-}\KeywordTok{grepl}\NormalTok{(}\StringTok{"2.cover"}\NormalTok{, }\KeywordTok{names}\NormalTok{(genotype_split))}
\NormalTok{genotype_cols<-}\KeywordTok{grepl}\NormalTok{(}\StringTok{"1.geno"}\NormalTok{, }\KeywordTok{names}\NormalTok{(genotype_split))}
\NormalTok{coverage<-genotype_split[coverage_cols]}
\NormalTok{genotype<-}\StringTok{ }\KeywordTok{read.delim}\NormalTok{(}\StringTok{"genotype.01.txt"}\NormalTok{, }\DataTypeTok{sep =} \StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\DataTypeTok{header =} \OtherTok{TRUE}\NormalTok{)}
\NormalTok{t.coverage<-}\KeywordTok{t}\NormalTok{(}\KeywordTok{as.matrix}\NormalTok{(coverage))}
\NormalTok{t.genotype<-}\KeywordTok{t}\NormalTok{(}\KeywordTok{as.matrix}\NormalTok{(genotype))}


 \CommentTok{# df3 = merge(t.coverage, t.genotype, by.x=row.names(t.coverage), by.y=row.names(t.genotype))}

\CommentTok{# Applying Ward Hierarchical Clustering}
\NormalTok{d_coverage   <-}\StringTok{ }\KeywordTok{dist}\NormalTok{(t.coverage, }\DataTypeTok{method=}\StringTok{"euclidean"}\NormalTok{)}
\NormalTok{ward_fit_coverage <-}\StringTok{ }\KeywordTok{hclust}\NormalTok{(d_coverage, }\DataTypeTok{method=}\StringTok{"ward"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## The "ward" method has been renamed to "ward.D"; note new "ward.D2"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{g4_ward_coverage <-}\StringTok{ }\KeywordTok{cutree}\NormalTok{(ward_fit_coverage, }\DataTypeTok{k =} \KeywordTok{c}\NormalTok{(}\DecValTok{4}\NormalTok{))}
\NormalTok{g8_ward_coverage <-}\StringTok{ }\KeywordTok{cutree}\NormalTok{(ward_fit_coverage, }\DataTypeTok{k =} \KeywordTok{c}\NormalTok{(}\DecValTok{8}\NormalTok{))}

\CommentTok{#using Gower via daisy}
\NormalTok{gower_dist_genotype <-}\StringTok{ }\KeywordTok{daisy}\NormalTok{(t.genotype,}
                    \DataTypeTok{metric =} \StringTok{"gower"}\NormalTok{,}
                    \DataTypeTok{type =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{logratio =} \DecValTok{3}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in daisy(t.genotype, metric = "gower", type = list(logratio = 3)):
## binary variable(s) 2, 4, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 17, 18, 19,
## 20, 21, 22, 23, 24, 25, 26 treated as interval scaled
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gower_mat_genotype <-}\StringTok{ }\KeywordTok{as.matrix}\NormalTok{(gower_dist_genotype)}


\NormalTok{tsne_obj_genotype <-}\StringTok{ }\KeywordTok{Rtsne}\NormalTok{(gower_dist_genotype, }\DataTypeTok{is_distance =} \OtherTok{TRUE}\NormalTok{)}
\NormalTok{pam_fit_genotype <-}\StringTok{ }\KeywordTok{pam}\NormalTok{(gower_dist_genotype, }\DataTypeTok{diss =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{k =} \DecValTok{3}\NormalTok{)}
\NormalTok{tsne_data_genotype <-}\StringTok{ }\NormalTok{tsne_obj_genotype$Y %>%}
\StringTok{  }\KeywordTok{data.frame}\NormalTok{() %>%}
\StringTok{  }\KeywordTok{setNames}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"X"}\NormalTok{, }\StringTok{"Y"}\NormalTok{)) %>%}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{cluster =} \KeywordTok{factor}\NormalTok{(pam_fit_genotype$clustering),}
         \DataTypeTok{name =} \KeywordTok{row.names}\NormalTok{(t.genotype))}


\CommentTok{# Gower clusters for coverage:}

\NormalTok{gower_dist_coverage <-}\StringTok{ }\KeywordTok{daisy}\NormalTok{(t.coverage,}
                    \DataTypeTok{metric =} \StringTok{"gower"}\NormalTok{,}
                    \DataTypeTok{type =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{logratio =} \DecValTok{3}\NormalTok{))}
\NormalTok{gower_mat_coverage <-}\StringTok{ }\KeywordTok{as.matrix}\NormalTok{(gower_dist_coverage)}


\NormalTok{tsne_obj_coverage <-}\StringTok{ }\KeywordTok{Rtsne}\NormalTok{(gower_dist_coverage, }\DataTypeTok{is_distance =} \OtherTok{TRUE}\NormalTok{)}
\NormalTok{pam_fit_coverage <-}\StringTok{ }\KeywordTok{pam}\NormalTok{(gower_dist_coverage, }\DataTypeTok{diss =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{k =} \DecValTok{3}\NormalTok{)}
\NormalTok{tsne_data_coverage <-}\StringTok{ }\NormalTok{tsne_obj_coverage$Y %>%}
\StringTok{  }\KeywordTok{data.frame}\NormalTok{() %>%}
\StringTok{  }\KeywordTok{setNames}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"X"}\NormalTok{, }\StringTok{"Y"}\NormalTok{)) %>%}
\StringTok{  }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{cluster =} \KeywordTok{factor}\NormalTok{(pam_fit_coverage$clustering),}
         \DataTypeTok{name =} \KeywordTok{row.names}\NormalTok{(t.coverage))}
\end{Highlighting}
\end{Shaded}

Writing data out

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{write.table}\NormalTok{(tsne_data_genotype, }\DataTypeTok{file=}\StringTok{"genotype_clusters.csv"}\NormalTok{, }\DataTypeTok{quote =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{sep =} \StringTok{","}\NormalTok{)}
\KeywordTok{write.table}\NormalTok{(g8_ward_coverage, }\DataTypeTok{file=}\StringTok{"coverage_groups.csv"}\NormalTok{, }\DataTypeTok{quote =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{sep =} \StringTok{","}\NormalTok{)}
\KeywordTok{write.table}\NormalTok{(tsne_data_coverage, }\DataTypeTok{file =} \StringTok{"coverage_clusters.csv"}\NormalTok{, }\DataTypeTok{quote =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{sep =} \StringTok{","}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Plots: Ward clustered dendrogram for coverage:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{plot}\NormalTok{(ward_fit_coverage)}
\end{Highlighting}
\end{Shaded}

\includegraphics{vcf_clustering_files/figure-latex/unnamed-chunk-7-1.pdf}

Genotype clusters:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{ggplot}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =} \NormalTok{X, }\DataTypeTok{y =} \NormalTok{Y), }\DataTypeTok{data =} \NormalTok{tsne_data_genotype) +}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{color =} \NormalTok{cluster))}
\end{Highlighting}
\end{Shaded}

\includegraphics{vcf_clustering_files/figure-latex/unnamed-chunk-8-1.pdf}

Samples in genotype clusters:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tsne_data_genotype}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##              X           Y cluster          name
## 1   -9.0275969 -6.24022771       1 P1_A01.1.geno
## 2   -8.5733970 -4.96771651       1 P1_A02.1.geno
## 3    2.9268395 -0.15391249       2 P1_A03.1.geno
## 4    3.6549049  2.75365209       2 P1_A04.1.geno
## 5    2.5982498 -0.83326148       2 P1_A05.1.geno
## 6    4.8746777  2.46090491       2 P1_A06.1.geno
## 7    4.1394616  1.04698684       2 P1_A07.1.geno
## 8   -3.6982637 -0.84295930       3 P1_A08.1.geno
## 9   -3.6982637 -0.84295930       3 P1_A09.1.geno
## 10  -8.5804480 -4.79813955       1 P1_A10.1.geno
## 11   3.1491796  3.92601472       2 P1_A11.1.geno
## 12  -3.6905131 -1.06522763       3 P1_A12.1.geno
## 13  -9.0256390 -6.23994384       1 P1_B01.1.geno
## 14   0.9942077  2.20661849       2 P1_B02.1.geno
## 15   1.3313714 -0.58053000       2 P1_B03.1.geno
## 16  -3.6905887 -1.06546154       3 P1_B04.1.geno
## 17   4.1349555  1.54191831       2 P1_B05.1.geno
## 18   1.2295602  2.70242443       2 P1_B06.1.geno
## 19   4.0070159  2.19271063       2 P1_B07.1.geno
## 20   1.5648114  0.37355042       2 P1_B08.1.geno
## 21   3.7217895  3.82947266       2 P1_B09.1.geno
## 22   3.7217895  3.82947266       2 P1_B10.1.geno
## 23   3.3431274  3.07648052       2 P1_B11.1.geno
## 24   3.9702056  2.27716388       2 P1_B12.1.geno
## 25   3.1376562 -0.95601103       2 P1_C01.1.geno
## 26  -9.0266415 -6.24052710       1 P1_C02.1.geno
## 27   0.7649760  3.37527338       2 P1_C03.1.geno
## 28   0.9872588  2.17177043       2 P1_C04.1.geno
## 29   2.8293319 -1.01990655       2 P1_C05.1.geno
## 30  -9.0260767 -6.24097859       1 P1_C06.1.geno
## 31   3.9788116  2.26280689       2 P1_C07.1.geno
## 32  -3.6660938 -0.89296383       3 P1_C08.1.geno
## 33  -9.0280735 -6.24084313       1 P1_C09.1.geno
## 34   2.3685632  1.91336634       2 P1_C10.1.geno
## 35   1.2263342  3.69217034       2 P1_C11.1.geno
## 36   1.3373933  2.80618602       2 P1_C12.1.geno
## 37  -9.0273296 -6.24062574       1 P1_D01.1.geno
## 38  -9.0275712 -6.24024982       1 P1_D02.1.geno
## 39   3.1114720 -0.16090742       2 P1_D03.1.geno
## 40   2.7847014  3.16177606       2 P1_D04.1.geno
## 41   2.5525417  3.98594002       2 P1_D05.1.geno
## 42   0.8795029  1.35354054       2 P1_D06.1.geno
## 43   2.1650138  3.96485462       2 P1_D07.1.geno
## 44  -9.3112805 -6.53371156       1 P1_D08.1.geno
## 45   0.9232555 -0.07103359       2 P1_D09.1.geno
## 46  -9.0273519 -6.24033613       1 P1_D11.1.geno
## 47  -8.5801521 -4.79810013       1 P1_D12.1.geno
## 48   4.6184305 -0.08891616       2 P1_E01.1.geno
## 49   2.2158735 -1.09501306       2 P1_E02.1.geno
## 50   1.5689882  2.95338811       2 P1_E03.1.geno
## 51   4.2704512  2.87709581       2 P1_E04.1.geno
## 52   1.7296677 -0.88897380       2 P1_E05.1.geno
## 53  -8.5762420 -4.96768769       1 P1_E06.1.geno
## 54   3.3055005 -0.12261126       2 P1_E07.1.geno
## 55   0.7499287  0.55361308       2 P1_E08.1.geno
## 56  -8.5735173 -4.96691769       1 P1_E09.1.geno
## 57   3.7310858 -0.76954903       2 P1_E11.1.geno
## 58   1.0489636 -0.16696390       2 P1_E12.1.geno
## 59   1.8737307  3.18954825       2 P1_F01.1.geno
## 60  -8.0880411 -6.24614431       1 P1_F02.1.geno
## 61   1.3146887 -0.40600607       2 P1_F03.1.geno
## 62   2.3346109 -0.07658695       2 P1_F04.1.geno
## 63   0.4630021  2.87624960       2 P1_F05.1.geno
## 64   2.1188688  3.14858559       2 P1_F06.1.geno
## 65   2.5579782 -0.14518007       2 P1_F07.1.geno
## 66  -7.8278075 -5.54217765       1 P1_F08.1.geno
## 67   2.0677735  1.58309762       2 P1_F09.1.geno
## 68   4.1397925  1.55229791       2 P1_F10.1.geno
## 69   2.7224988 -0.26913181       2 P1_F11.1.geno
## 70   0.3131316  1.94944681       2 P1_F12.1.geno
## 71  -8.9373169 -6.42994699       1 P1_G02.1.geno
## 72   4.3219183  3.47435494       2 P1_G03.1.geno
## 73   4.0429056  2.10404789       2 P1_G04.1.geno
## 74  -9.0247932 -6.23955247       1 P1_G05.1.geno
## 75  -3.9515518 -1.89563858       3 P1_G06.1.geno
## 76   4.3575570  3.48407411       2 P1_G07.1.geno
## 77   3.7314779 -0.76938854       2 P1_G08.1.geno
## 78   1.8774086  0.10978213       2 P1_G09.1.geno
## 79   4.3300604  0.87076909       2 P1_G10.1.geno
## 80  -9.0277630 -6.24046365       1 P1_G11.1.geno
## 81   4.3154816 -0.38037783       2 P1_G12.1.geno
## 82   1.6369589  0.33722983       2 P1_H01.1.geno
## 83   4.0202323  2.15109055       2 P1_H02.1.geno
## 84   2.0255979  0.01666963       2 P1_H03.1.geno
## 85   4.1487730  1.51988551       2 P1_H04.1.geno
## 86   4.6797385  2.59496588       2 P1_H05.1.geno
## 87   5.0076614  2.33669893       2 P1_H06.1.geno
## 88  -9.0270988 -6.24036010       1 P1_H07.1.geno
## 89   2.5639875  1.95336536       2 P1_H08.1.geno
## 90   3.6552628  2.75425953       2 P1_H09.1.geno
## 91   4.1497818  1.53310848       2 P1_H10.1.geno
## 92   3.2088069 -0.14482005       2 P1_H11.1.geno
## 93   2.7217290  3.17275003       2 P1_H12.1.geno
## 94   4.7029228  0.09445591       2 P2_A01.1.geno
## 95   2.6144339  3.12398529       2 P2_A02.1.geno
## 96   3.7762785 -0.01382083       2 P2_A03.1.geno
## 97   2.7475669  3.12306632       2 P2_A04.1.geno
## 98  -8.6287420 -4.76269251       1 P2_A05.1.geno
## 99   1.4802740  3.85705300       2 P2_A07.1.geno
## 100  0.7416349  1.77484588       2 P2_A08.1.geno
## 101  5.0896634  1.55863219       2 P2_A09.1.geno
## 102  2.0079887  3.15781604       2 P2_A10.1.geno
## 103  1.2648344  0.66580522       2 P2_A11.1.geno
## 104  2.5279215 -0.09189322       2 P2_A12.1.geno
## 105 -8.5728273 -4.96809034       1 P2_B01.1.geno
## 106  1.4263310  2.79706384       2 P2_B02.1.geno
## 107  0.8375271  1.49463950       2 P2_B06.1.geno
## 108 -3.6905474 -1.06551905       3 P2_B07.1.geno
## 109  1.0686268  2.30991037       2 P2_B08.1.geno
## 110  5.1579755  0.91296430       2 P2_B09.1.geno
## 111  2.9717290  1.44468930       2 P2_B10.1.geno
## 112  2.9240073  1.80293333       2 P2_B11.1.geno
## 113  3.1477670  1.52029401       2 P2_B12.1.geno
## 114  1.4096462  0.51150246       2 P2_C02.1.geno
## 115 -0.7491210  0.37500260       2 P2_C03.1.geno
## 116 -1.0776133  0.24301716       3 P2_C05.1.geno
## 117 -0.8751639  0.32482837       2 P2_C07.1.geno
## 118  2.9696054  1.74972021       2 P2_C08.1.geno
## 119  2.8801045  1.85067619       2 P2_C09.1.geno
## 120  3.1079887  1.57243056       2 P2_C12.1.geno
## 121  3.0500855  1.61074713       2 P2_D03.1.geno
## 122  3.0240369  1.67103113       2 P2_D04.1.geno
## 123  2.8801044  1.85067633       2 P2_D05.1.geno
## 124 -9.0270405 -6.24018859       1 P2_D06.1.geno
## 125 -9.0281172 -6.24009170       1 P2_D07.1.geno
## 126 -9.0274075 -6.23958213       1 P2_D08.1.geno
## 127  3.0500855  1.61074713       2 P2_D09.1.geno
## 128 -9.0264096 -6.24011622       1 P2_E01.1.geno
## 129  2.2744632  1.83348594       2 P2_E02.1.geno
## 130  2.8801044  1.85067633       2 P2_E03.1.geno
## 131 -3.6905787 -1.06555795       3 P2_E05.1.geno
## 132  0.3703816  1.41633192       2 P2_E06.1.geno
## 133 -9.3140770 -6.57432986       1 P2_E09.1.geno
## 134  2.8586769  1.62368343       2 P2_E11.1.geno
## 135  2.7507631  1.91377530       2 P2_E12.1.geno
## 136 -9.0282985 -6.24013499       1 P2_F02.1.geno
## 137  2.8083848  1.89093691       2 P2_F03.1.geno
## 138 -3.6905787 -1.06555798       3 P2_F04.1.geno
## 139  2.5510461  1.95719568       2 P2_F05.1.geno
## 140  2.8759021  1.28507992       2 P2_F06.1.geno
## 141  2.5239082  1.94794736       2 P2_F08.1.geno
## 142  2.8069071  1.22124777       2 P2_F09.1.geno
## 143  2.8759021  1.28507992       2 P2_G03.1.geno
## 144  4.1341917  2.96918304       2 P2_G05.1.geno
## 145  2.3072956  1.65591711       2 P2_G06.1.geno
## 146 -9.0279052 -6.23964932       1 P2_G09.1.geno
## 147  2.4155908  1.15556826       2 P2_H01.1.geno
## 148  2.4097797  1.15128897       2 P2_H02.1.geno
## 149  2.3056422  1.65361846       2 P2_H04.1.geno
## 150  2.3225646  1.52098758       2 P2_H05.1.geno
## 151  2.5658402  1.94999778       2 P2_H06.1.geno
## 152  2.3685634  1.91336642       2 P2_H07.1.geno
## 153  2.7899187  1.20341811       2 P2_H08.1.geno
## 154  2.0691820  1.57483365       2 P2_H09.1.geno
## 155 -8.5735094 -4.96780277       1 P3_A01.1.geno
## 156 -7.9330737 -4.25241156       3 P3_A02.1.geno
## 157  2.8016030  1.21507996       2 P3_A03.1.geno
## 158 -9.0277258 -6.24036628       1 P3_A04.1.geno
## 159  2.7916814  1.20607991       2 P3_A05.1.geno
## 160  2.0611301  1.53768271       2 P3_A06.1.geno
## 161  2.0617646  1.57769428       2 P3_A07.1.geno
## 162  2.7963708  1.21031253       2 P3_A08.1.geno
## 163  2.5196929  1.94290661       2 P3_A09.1.geno
## 164 -9.0327126 -6.24055727       1 P3_A10.1.geno
## 165  2.4872640  1.93737460       2 P3_A11.1.geno
## 166  2.0604123  1.51662385       2 P3_A12.1.geno
## 167  2.7900810  1.20355151       2 P3_B01.1.geno
## 168  2.7912845  1.20574883       2 P3_B02.1.geno
## 169 -9.0279131 -6.24066640       1 P3_B03.1.geno
## 170  2.3780718  1.18039530       2 P3_B04.1.geno
## 171  2.3802405  1.17681375       2 P3_B05.1.geno
## 172  2.3011277  1.65867205       2 P3_B06.1.geno
## 173  2.3940047  1.16786437       2 P3_B07.1.geno
## 174  2.3041390  1.67022258       2 P3_B08.1.geno
## 175  2.3828779  1.17491536       2 P3_B09.1.geno
## 176  2.7911646  1.20579015       2 P3_B10.1.geno
## 177  2.3984600  1.16567284       2 P3_B11.1.geno
## 178  2.1028964  1.58480396       2 P3_B12.1.geno
## 179 -8.5746671 -4.96724385       1 P3_C01.1.geno
## 180 -8.5736302 -4.96765507       1 P3_C02.1.geno
## 181  2.3778296  1.17853690       2 P3_C03.1.geno
## 182  2.7897206  1.20324511       2 P3_C04.1.geno
## 183  2.3726923  1.18570770       2 P3_C05.1.geno
## 184 -9.0273011 -6.24008951       1 P3_C06.1.geno
## 185  2.3110630  1.70095964       2 P3_C07.1.geno
## 186  2.4934746  1.93886644       2 P3_C08.1.geno
## 187  2.7885075  1.20212452       2 P3_C09.1.geno
## 188  2.5017396  1.93889452       2 P3_C10.1.geno
## 189  2.4025124  1.16264517       2 P3_D01.1.geno
## 190  2.3752404  1.18019721       2 P3_D02.1.geno
## 191  2.7883642  1.20219369       2 P3_D03.1.geno
## 192  2.4101002  1.15785168       2 P3_D05.1.geno
## 193  2.3044205  1.64603358       2 P3_D06.1.geno
## 194  2.7881881  1.20193354       2 P3_D07.1.geno
## 195  2.3750859  1.90939091       2 P3_D08.1.geno
## 196  2.0677735  1.58309763       2 P3_D09.1.geno
## 197  2.3788172  1.17765871       2 P3_D11.1.geno
## 198  2.0605935  1.51482458       2 P3_D12.1.geno
## 199  0.4989277  0.88597621       2 P3_E02.1.geno
## 200  2.0677735  1.58309763       2 P3_E03.1.geno
## 201 -8.4019993 -6.27575317       1 P3_E04.1.geno
## 202  2.0558517  1.49582135       2 P3_E05.1.geno
## 203  2.3773102  1.17891834       2 P3_E06.1.geno
## 204  2.7904144  1.20513854       2 P3_E07.1.geno
## 205 -7.6445891 -5.43026810       1 P3_F01.1.geno
## 206 -9.0278578 -6.24037685       1 P3_F04.1.geno
## 207  2.7900810  1.20355151       2 P3_F05.1.geno
## 208  2.7921867  1.20792339       2 P3_F08.1.geno
## 209  2.3072956  1.65591711       2 P3_F12.1.geno
## 210  2.7967705  1.21113663       2 P3_G02.1.geno
## 211  2.7938769  1.20784580       2 P3_G03.1.geno
## 212  2.3685412  1.91335675       2 P3_G04.1.geno
## 213  2.8202966  1.88060830       2 P3_G05.1.geno
## 214  2.3782135  1.18147777       2 P3_G06.1.geno
## 215  2.9057950  1.82422912       2 P3_H05.1.geno
## 216 -8.5743118 -4.96743109       1 P4_A01.1.geno
## 217 -3.6906417 -1.06553534       3 P4_A02.1.geno
## 218  3.0500855  1.61074713       2 P4_A03.1.geno
## 219 -8.5745503 -4.96773118       1 P4_A07.1.geno
## 220 -0.6263837  1.62647795       1 P4_A08.1.geno
## 221 -8.5745524 -4.96785098       1 P4_A09.1.geno
## 222  2.9626916  1.75981460       2 P4_A10.1.geno
## 223 -3.9513876 -1.89613851       3 P4_A11.1.geno
## 224  2.8415762  1.86822298       2 P4_A12.1.geno
## 225 -3.6905950 -1.06553352       3 P4_B01.1.geno
## 226 -8.5739543 -4.96756009       1 P4_B02.1.geno
## 227 -8.5734954 -4.96735406       1 P4_B03.1.geno
## 228 -3.6905765 -1.06559958       3 P4_B05.1.geno
## 229  3.0500855  1.61074713       2 P4_B06.1.geno
## 230  2.8586769  1.62368343       2 P4_B07.1.geno
## 231  3.0500855  1.61074713       2 P4_B08.1.geno
## 232  2.9595015  1.76359907       2 P4_B09.1.geno
## 233  2.3056435  1.65362293       2 P4_B11.1.geno
## 234  4.9702665  1.10859804       2 P4_B12.1.geno
## 235  1.7749188  0.17313048       2 P4_C03.1.geno
## 236 -9.0275579 -6.24041697       1 P4_C04.1.geno
## 237  0.8787586  1.35096512       2 P4_C05.1.geno
## 238 -0.7491210  0.37500260       2 P4_C06.1.geno
## 239 -3.6904313 -1.06555768       3 P4_C07.1.geno
## 240  0.7552745  0.64838149       2 P4_C08.1.geno
## 241  5.0236349  0.64998647       2 P4_C10.1.geno
## 242 -3.6904160 -1.06571476       3 P4_C11.1.geno
## 243  3.5771857 -0.08810084       2 P4_D01.1.geno
## 244  1.9019138 -0.66659642       2 P4_D02.1.geno
## 245  1.2091785  0.83097429       2 P4_D03.1.geno
## 246 -8.5751149 -4.96763257       1 P4_D04.1.geno
## 247  2.8020909  3.16361634       2 P4_D05.1.geno
## 248 -3.6660946 -0.89296444       3 P4_D06.1.geno
## 249  2.0569492  0.01530520       2 P4_D07.1.geno
## 250 -3.6905787 -1.06555795       3 P4_D08.1.geno
## 251  4.2385135  0.87073660       2 P4_D09.1.geno
## 252 -3.6905787 -1.06555795       3 P4_D11.1.geno
## 253 -9.0269296 -6.24022086       1 P4_D12.1.geno
## 254  4.2385175  0.87084449       2 P4_E01.1.geno
## 255 -8.5735148 -4.96710751       1 P4_E02.1.geno
## 256 -3.6906666 -1.06556599       3 P4_E03.1.geno
## 257 -3.5817761 -1.67893252       3 P4_E04.1.geno
## 258  1.8618528  3.09287285       2 P4_E05.1.geno
## 259 -3.8635711 -1.98807280       3 P4_E06.1.geno
## 260 -3.6906665 -1.06556600       3 P4_E07.1.geno
## 261  1.0247974  2.43430961       2 P4_E08.1.geno
## 262 -3.6886429 -1.06615687       3 P4_E09.1.geno
## 263  1.3668937  2.75336571       2 P4_E11.1.geno
## 264  1.8162512  3.18445138       2 P4_E12.1.geno
## 265  1.2054666  0.80991627       2 P4_F01.1.geno
## 266 -3.6905787 -1.06555798       3 P4_F02.1.geno
## 267 -3.9513908 -1.89646382       3 P4_F03.1.geno
## 268  5.1095536  1.72668421       2 P4_F04.1.geno
## 269  0.3165625  2.43321308       2 P4_F05.1.geno
## 270 -8.5740386 -4.96785630       1 P4_F06.1.geno
## 271 -3.5817834 -1.67893295       3 P4_F07.1.geno
## 272  0.8090803  0.48165025       2 P4_F08.1.geno
## 273  2.2746997  3.91637675       2 P4_F09.1.geno
## 274  0.9803535  1.39106430       2 P4_F10.1.geno
## 275  4.4735213  1.72886380       2 P4_F12.1.geno
## 276  3.9009319  0.41285273       2 P4_G01.1.geno
## 277  0.9462675  2.07131684       2 P4_G07.1.geno
## 278  3.0255228  3.98400414       2 P4_G09.1.geno
## 279  0.8333512  1.49901766       2 P4_G11.1.geno
## 280  3.6503445  2.79681279       2 P4_H01.1.geno
## 281  3.3376689  3.07181055       2 P4_H06.1.geno
## 282  3.3376689  3.07181055       2 P4_H07.1.geno
## 283  3.9008000  0.41264478       2 P4_H09.1.geno
## 284  3.9007959  0.41263820       2 P4_H10.1.geno
## 285  3.9007959  0.41263820       2 P4_H11.1.geno
## 286  2.3768538  1.91415786       2 P4_H12.1.geno
\end{verbatim}

Coverage Gower clusters:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{ggplot}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x =} \NormalTok{X, }\DataTypeTok{y =} \NormalTok{Y), }\DataTypeTok{data =} \NormalTok{tsne_data_coverage) +}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{(}\KeywordTok{aes}\NormalTok{(}\DataTypeTok{color =} \NormalTok{cluster))}
\end{Highlighting}
\end{Shaded}

\includegraphics{vcf_clustering_files/figure-latex/unnamed-chunk-10-1.pdf}

Samples and coverage clusters:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tsne_data_coverage}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                X           Y cluster           name
## 1   -1.707840350  4.03834336       1 P1_A01.2.cover
## 2   -3.234992609  3.14827738       2 P1_A02.2.cover
## 3    3.039817477 -3.94055846       1 P1_A03.2.cover
## 4    7.325071292 -3.35182140       1 P1_A04.2.cover
## 5    5.710366432  1.40212511       1 P1_A05.2.cover
## 6    1.833646178 -0.75891240       1 P1_A06.2.cover
## 7    0.367150596  3.05775391       1 P1_A07.2.cover
## 8   -1.666797962  1.52124734       1 P1_A08.2.cover
## 9   -6.318301325  0.05836090       1 P1_A09.2.cover
## 10  -3.221169991  2.01537725       1 P1_A10.2.cover
## 11  -7.638151684  0.27925837       2 P1_A11.2.cover
## 12  -1.514381857  1.86506440       1 P1_A12.2.cover
## 13  -8.934096809  0.90856027       2 P1_B01.2.cover
## 14   2.387642011 -2.83243204       1 P1_B02.2.cover
## 15  -7.003830930 -0.19374150       1 P1_B03.2.cover
## 16  -7.366688131  2.17982069       2 P1_B04.2.cover
## 17  -0.537488560 -2.25101825       1 P1_B05.2.cover
## 18   6.285654153  1.18126430       1 P1_B06.2.cover
## 19   5.536892639 -4.55904451       1 P1_B07.2.cover
## 20   2.171737911  0.17956336       1 P1_B08.2.cover
## 21   4.891868799  2.49788358       1 P1_B09.2.cover
## 22  -1.949980688 -6.96665298       1 P1_B10.2.cover
## 23   6.658755751  1.89627962       1 P1_B11.2.cover
## 24   7.004624099  0.44575644       1 P1_B12.2.cover
## 25   4.119182667 -2.22262197       1 P1_C01.2.cover
## 26  -9.169146189  1.12915081       2 P1_C02.2.cover
## 27   0.171600587  2.00607204       1 P1_C03.2.cover
## 28   7.374800890  1.23887767       1 P1_C04.2.cover
## 29  -7.226301878 -0.04326171       1 P1_C05.2.cover
## 30  -6.388231227  3.21680088       2 P1_C06.2.cover
## 31  -0.543451774 -2.23057390       1 P1_C07.2.cover
## 32  -8.863502736  3.10138181       3 P1_C08.2.cover
## 33   0.184457553  5.10417649       1 P1_C09.2.cover
## 34   6.295409051 -3.14117847       1 P1_C10.2.cover
## 35   6.527483336 -4.01528248       1 P1_C11.2.cover
## 36   0.181898189  2.02126616       1 P1_C12.2.cover
## 37  -7.216101607  3.32173068       2 P1_D01.2.cover
## 38  -1.051474192  4.05820200       1 P1_D02.2.cover
## 39  -0.544718503 -2.23683259       1 P1_D03.2.cover
## 40   5.852645046  2.28193015       1 P1_D04.2.cover
## 41   2.169170195 -2.39262815       1 P1_D05.2.cover
## 42  -8.087542032  0.88341754       2 P1_D06.2.cover
## 43  -6.796870451  0.01157209       1 P1_D07.2.cover
## 44   0.184415996  5.10416667       1 P1_D08.2.cover
## 45  -6.723291701 -0.27646047       1 P1_D09.2.cover
## 46   0.282602397  5.10094242       1 P1_D11.2.cover
## 47  -7.052281297  2.18931331       2 P1_D12.2.cover
## 48   4.766377193 -0.69900038       1 P1_E01.2.cover
## 49   5.042642127 -1.53170986       1 P1_E02.2.cover
## 50  -0.019335221  5.08049763       1 P1_E03.2.cover
## 51   0.282633136  5.10095576       1 P1_E04.2.cover
## 52   5.407289883 -2.11974720       1 P1_E05.2.cover
## 53  -3.868262173  3.37980601       1 P1_E06.2.cover
## 54  -2.510435332  0.71218851       1 P1_E07.2.cover
## 55  -2.936492302  0.70974694       1 P1_E08.2.cover
## 56  -3.150881960  2.08780981       1 P1_E09.2.cover
## 57   5.020486823 -2.27530189       1 P1_E11.2.cover
## 58   5.754606637 -0.34415094       1 P1_E12.2.cover
## 59   5.952296088 -1.33307691       1 P1_F01.2.cover
## 60  -6.812929528  3.52397731       2 P1_F02.2.cover
## 61  -9.433693162  2.95410189       3 P1_F03.2.cover
## 62   0.330787831  5.08766012       1 P1_F04.2.cover
## 63   4.548841481 -2.37155848       1 P1_F05.2.cover
## 64   5.828292290 -0.76958874       1 P1_F06.2.cover
## 65   5.836370807 -0.64416293       1 P1_F07.2.cover
## 66  -3.205192072  3.58180644       2 P1_F08.2.cover
## 67   4.735662787 -2.35899187       1 P1_F09.2.cover
## 68  -2.473004445  0.66966351       1 P1_F10.2.cover
## 69   5.814325604 -0.87000617       1 P1_F11.2.cover
## 70  -0.007968634  1.94887560       1 P1_F12.2.cover
## 71  -3.556155500  2.73386879       1 P1_G02.2.cover
## 72  -9.510282387  0.19143728       3 P1_G03.2.cover
## 73   4.133402264 -0.42786895       1 P1_G04.2.cover
## 74  -9.201962667  1.10212364       2 P1_G05.2.cover
## 75  -3.050875322  2.90046082       2 P1_G06.2.cover
## 76  -2.172755727 -6.76977178       1 P1_G07.2.cover
## 77   0.170324021  3.43634864       1 P1_G08.2.cover
## 78  -9.080423220  2.58198186       3 P1_G09.2.cover
## 79   4.806919375 -0.63017205       1 P1_G10.2.cover
## 80  -9.066171187  0.91932483       2 P1_G11.2.cover
## 81   4.617227031  0.26045534       1 P1_G12.2.cover
## 82   4.689632139  0.24263178       1 P1_H01.2.cover
## 83   5.695722971 -0.19251691       1 P1_H02.2.cover
## 84  -3.167761897  1.49709422       1 P1_H03.2.cover
## 85   4.743561988  0.21578219       1 P1_H04.2.cover
## 86   4.117246182 -2.03241087       1 P1_H05.2.cover
## 87   4.886025712 -1.59270980       1 P1_H06.2.cover
## 88  -8.538719950  0.94025981       2 P1_H07.2.cover
## 89  -7.226301878 -0.04326171       1 P1_H08.2.cover
## 90   3.879691730  0.13322277       1 P1_H09.2.cover
## 91  -0.546009639 -2.25481551       1 P1_H10.2.cover
## 92  -0.541904884 -2.25243470       1 P1_H11.2.cover
## 93   5.556254501 -1.17369671       1 P1_H12.2.cover
## 94   3.661289497  0.07788003       1 P2_A01.2.cover
## 95   1.696580905 -1.71484008       1 P2_A02.2.cover
## 96   3.654122634 -0.62143467       1 P2_A03.2.cover
## 97   0.167766623  1.99181853       1 P2_A04.2.cover
## 98  -3.893823086  3.28176663       1 P2_A05.2.cover
## 99   4.457235165 -1.19081421       1 P2_A07.2.cover
## 100 -0.541976604 -2.25240604       1 P2_A08.2.cover
## 101  3.511462891 -0.70741780       1 P2_A09.2.cover
## 102  5.616934895 -0.01926867       1 P2_A10.2.cover
## 103  3.853782262 -2.04747195       1 P2_A11.2.cover
## 104  4.987833081 -0.28009664       1 P2_A12.2.cover
## 105 -7.761657573  1.86242692       2 P2_B01.2.cover
## 106 -0.545601945 -2.24933675       1 P2_B02.2.cover
## 107  4.854604062 -1.49127305       1 P2_B06.2.cover
## 108 -3.329147964  1.40276527       1 P2_B07.2.cover
## 109  5.100293606 -0.26540521       1 P2_B08.2.cover
## 110  0.235611294  5.10350044       1 P2_B09.2.cover
## 111  4.130706975  0.06648983       1 P2_B10.2.cover
## 112  3.671309485 -1.12010787       1 P2_B11.2.cover
## 113  3.301184458 -0.35668087       1 P2_B12.2.cover
## 114  5.290251985 -1.32416502       1 P2_C02.2.cover
## 115 -1.870668174 -7.03981040       1 P2_C03.2.cover
## 116 -8.947866730  3.11848965       3 P2_C05.2.cover
## 117 -7.073413564 -0.78880697       1 P2_C07.2.cover
## 118  3.198207012 -0.90946990       1 P2_C08.2.cover
## 119  3.660012561  0.11508253       1 P2_C09.2.cover
## 120  4.068293702  0.12984530       1 P2_C12.2.cover
## 121  4.108287948 -0.55177591       1 P2_D03.2.cover
## 122  4.200876997 -1.04654714       1 P2_D04.2.cover
## 123 -2.367477378 -6.56450118       1 P2_D05.2.cover
## 124 -1.750700936  4.00413589       1 P2_D06.2.cover
## 125  0.282605741  5.10094386       1 P2_D07.2.cover
## 126 -6.356370197  0.76077969       1 P2_D08.2.cover
## 127  0.184426925  5.10416896       1 P2_D09.2.cover
## 128 -7.470725823  1.45069138       2 P2_E01.2.cover
## 129 -0.656785059  2.10357800       1 P2_E02.2.cover
## 130  4.070338241 -1.02726091       1 P2_E03.2.cover
## 131 -8.421625710  3.33994975       3 P2_E05.2.cover
## 132 -9.456289276  1.79922114       3 P2_E06.2.cover
## 133 -1.605019860  4.06050023       1 P2_E09.2.cover
## 134  4.120554564 -0.55518140       1 P2_E11.2.cover
## 135  3.353978492 -1.60802165       1 P2_E12.2.cover
## 136 -0.300700297  5.08643132       1 P2_F02.2.cover
## 137  4.065830082 -0.57793168       1 P2_F03.2.cover
## 138 -8.924115715  3.00467377       3 P2_F04.2.cover
## 139 -6.265892307  1.37462843       2 P2_F05.2.cover
## 140  0.171588787  2.00607404       1 P2_F06.2.cover
## 141 -2.957223571  0.64628305       1 P2_F08.2.cover
## 142 -1.870668174 -7.03981040       1 P2_F09.2.cover
## 143  3.938431582 -2.08490966       1 P2_G03.2.cover
## 144  7.407990202 -2.06924899       1 P2_G05.2.cover
## 145 -2.463149589 -6.52509016       1 P2_G06.2.cover
## 146 -7.873103120  3.86605628       3 P2_G09.2.cover
## 147  4.738294213 -2.25660532       1 P2_H01.2.cover
## 148  5.958880774 -1.42485182       1 P2_H02.2.cover
## 149 -1.870668132 -7.03981040       1 P2_H04.2.cover
## 150  5.950503216 -1.46663664       1 P2_H05.2.cover
## 151  5.350416943 -3.71886754       1 P2_H06.2.cover
## 152  5.426103665 -2.17805635       1 P2_H07.2.cover
## 153 -2.047457066 -6.88222200       1 P2_H08.2.cover
## 154  2.728299822 -3.12155423       1 P2_H09.2.cover
## 155 -3.863195661  4.15530407       2 P3_A01.2.cover
## 156 -7.130456377  2.33504539       2 P3_A02.2.cover
## 157  6.582876038  0.94974972       1 P3_A03.2.cover
## 158 -1.095418462  4.11455722       1 P3_A04.2.cover
## 159  4.092151627 -4.58675190       1 P3_A05.2.cover
## 160  2.559801395  0.71953487       1 P3_A06.2.cover
## 161  3.227516586 -3.31137754       1 P3_A07.2.cover
## 162  8.062863869  0.33255103       1 P3_A08.2.cover
## 163  6.707233129 -2.84960282       1 P3_A09.2.cover
## 164 -9.083187321  0.87077767       2 P3_A10.2.cover
## 165  3.773098179 -3.69158065       1 P3_A11.2.cover
## 166  0.208535402  2.06627506       1 P3_A12.2.cover
## 167 -2.367477378 -6.56450118       1 P3_B01.2.cover
## 168  5.396568418  0.10499241       1 P3_B02.2.cover
## 169  0.184426925  5.10416896       1 P3_B03.2.cover
## 170  4.183078948 -1.77932805       1 P3_B04.2.cover
## 171  4.755954608 -1.53659344       1 P3_B05.2.cover
## 172  5.459999204 -1.22249938       1 P3_B06.2.cover
## 173  3.451567837 -0.46323102       1 P3_B07.2.cover
## 174  4.810008904 -0.49004596       1 P3_B08.2.cover
## 175  4.792958422 -1.57826843       1 P3_B09.2.cover
## 176  4.920018404 -1.40729525       1 P3_B10.2.cover
## 177  3.495415114 -1.72520153       1 P3_B11.2.cover
## 178  4.148196420  0.21036847       1 P3_B12.2.cover
## 179 -8.520219015  4.38301972       3 P3_C01.2.cover
## 180 -7.850044299  1.61516634       2 P3_C02.2.cover
## 181  3.733932470 -1.30804409       1 P3_C03.2.cover
## 182  3.395474605 -1.52858682       1 P3_C04.2.cover
## 183  4.310147609 -1.08060937       1 P3_C05.2.cover
## 184  0.184457553  5.10417649       1 P3_C06.2.cover
## 185  3.161609348 -0.96527570       1 P3_C07.2.cover
## 186  5.161038741 -0.65422500       1 P3_C08.2.cover
## 187  3.673824518 -1.99979290       1 P3_C09.2.cover
## 188  4.836936485 -0.28611770       1 P3_C10.2.cover
## 189  3.331699337 -0.28946927       1 P3_D01.2.cover
## 190 -0.528420030 -2.25007815       1 P3_D02.2.cover
## 191  3.217065931 -0.88905688       1 P3_D03.2.cover
## 192  3.218901335 -1.39118878       1 P3_D05.2.cover
## 193  5.317691532 -0.68444768       1 P3_D06.2.cover
## 194  4.081705161 -0.73249238       1 P3_D07.2.cover
## 195  5.165546921 -0.67146150       1 P3_D08.2.cover
## 196  4.395666198 -1.87940460       1 P3_D09.2.cover
## 197 -1.870528556 -7.03978622       1 P3_D11.2.cover
## 198  5.314016063 -1.60260037       1 P3_D12.2.cover
## 199  3.352976714 -1.55663464       1 P3_E02.2.cover
## 200 -6.915327235  0.25834732       1 P3_E03.2.cover
## 201 -1.488315764  3.71231159       1 P3_E04.2.cover
## 202  4.444816391 -1.18629171       1 P3_E05.2.cover
## 203  4.117431554 -1.20415236       1 P3_E06.2.cover
## 204  4.413448238 -0.25398468       1 P3_E07.2.cover
## 205 -3.278869573  3.40049779       2 P3_F01.2.cover
## 206 -3.486564391  3.86706261       1 P3_F04.2.cover
## 207  5.317021125  0.11219130       1 P3_F05.2.cover
## 208  4.627016043  0.27770302       1 P3_F08.2.cover
## 209  5.156421327  0.18283254       1 P3_F12.2.cover
## 210  4.107670491 -1.47119009       1 P3_G02.2.cover
## 211  5.903972790 -1.13433619       1 P3_G03.2.cover
## 212  4.169611526 -2.11422303       1 P3_G04.2.cover
## 213  4.683049278 -2.36490824       1 P3_G05.2.cover
## 214  5.772598466 -0.78449020       1 P3_G06.2.cover
## 215  5.893376349 -1.75882637       1 P3_H05.2.cover
## 216 -6.673693357  3.26101811       2 P4_A01.2.cover
## 217 -8.088786271  3.75762679       3 P4_A02.2.cover
## 218 -2.367471191 -6.56451078       1 P4_A03.2.cover
## 219 -3.187887966  2.25530084       1 P4_A07.2.cover
## 220 -7.972560892  0.66933278       2 P4_A08.2.cover
## 221 -8.592413053  4.00001346       3 P4_A09.2.cover
## 222 -7.649677490 -0.34721853       2 P4_A10.2.cover
## 223 -6.657872577  3.28483448       2 P4_A11.2.cover
## 224  5.633676546 -1.98403644       1 P4_A12.2.cover
## 225 -8.648325438  2.89493485       3 P4_B01.2.cover
## 226 -8.094087327  2.02837431       2 P4_B02.2.cover
## 227 -8.314177300  4.06431011       3 P4_B03.2.cover
## 228 -1.664434675  1.95585304       1 P4_B05.2.cover
## 229  5.414396629 -2.18683228       1 P4_B06.2.cover
## 230 -2.554162355  0.76858777       1 P4_B07.2.cover
## 231  4.549564868 -3.78497317       1 P4_B08.2.cover
## 232 -6.735224799 -0.29452951       1 P4_B09.2.cover
## 233 -9.106659687  0.62627621       1 P4_B11.2.cover
## 234  2.949974429  1.20046682       1 P4_B12.2.cover
## 235  8.308253389 -1.02416256       1 P4_C03.2.cover
## 236 -3.020548537  2.86810965       2 P4_C04.2.cover
## 237  5.762956483 -3.63119338       1 P4_C05.2.cover
## 238 -1.870668174 -7.03981040       1 P4_C06.2.cover
## 239 -1.500208147  1.77467625       1 P4_C07.2.cover
## 240 -2.172755727 -6.76977178       1 P4_C08.2.cover
## 241  4.549564868 -3.78497317       1 P4_C10.2.cover
## 242 -8.907053358  3.30122071       3 P4_C11.2.cover
## 243 -7.249409585  2.22085092       2 P4_D01.2.cover
## 244  5.553431196 -2.10240547       1 P4_D02.2.cover
## 245 -3.019540198  3.29896131       1 P4_D03.2.cover
## 246 -5.716612332  2.30488784       2 P4_D04.2.cover
## 247  0.400012529  5.09897059       1 P4_D05.2.cover
## 248 -3.183230703  1.46167866       1 P4_D06.2.cover
## 249 -6.903864532 -0.24813245       1 P4_D07.2.cover
## 250 -9.478781527  3.52652506       3 P4_D08.2.cover
## 251  3.423082088  1.60923645       1 P4_D09.2.cover
## 252 -3.665854742  1.27122772       1 P4_D11.2.cover
## 253 -1.048396596  4.08563730       1 P4_D12.2.cover
## 254  3.894184340  1.64204664       1 P4_E01.2.cover
## 255 -3.642955145  3.54865456       2 P4_E02.2.cover
## 256 -7.236056758  1.41218763       2 P4_E03.2.cover
## 257 -7.757629461  1.59031775       2 P4_E04.2.cover
## 258 -9.246392855  0.96222057       2 P4_E05.2.cover
## 259 -3.520451871  0.72565709       1 P4_E06.2.cover
## 260 -8.316571401  2.27402663       2 P4_E07.2.cover
## 261 -2.462787599 -5.65900339       1 P4_E08.2.cover
## 262 -2.635217953  0.79860368       1 P4_E09.2.cover
## 263 -7.545664348 -0.04544016       2 P4_E11.2.cover
## 264  7.935675501 -2.46344953       1 P4_E12.2.cover
## 265  6.976254583  0.12268898       1 P4_F01.2.cover
## 266 -8.856653967  3.28983013       3 P4_F02.2.cover
## 267 -2.882376330  3.31541201       1 P4_F03.2.cover
## 268 -1.949952738 -6.96670953       1 P4_F04.2.cover
## 269 -2.883230918 -6.47876996       1 P4_F05.2.cover
## 270 -9.122954001  1.66245072       2 P4_F06.2.cover
## 271 -7.126333178  0.91408290       2 P4_F07.2.cover
## 272 -9.138183545  3.26248707       3 P4_F08.2.cover
## 273 -9.306469013  2.99494558       3 P4_F09.2.cover
## 274 -2.047477307 -6.88219257       1 P4_F10.2.cover
## 275  4.015189691  2.23919667       1 P4_F12.2.cover
## 276  4.945809936  1.61365979       1 P4_G01.2.cover
## 277  1.833646178 -0.75891240       1 P4_G07.2.cover
## 278  4.945809936  1.61365979       1 P4_G09.2.cover
## 279 -2.045580026 -6.88227317       1 P4_G11.2.cover
## 280  6.961004711 -2.29575377       1 P4_H01.2.cover
## 281  7.359777484 -0.45113175       1 P4_H06.2.cover
## 282  7.359777484 -0.45113175       1 P4_H07.2.cover
## 283 -1.949955026 -6.96670400       1 P4_H09.2.cover
## 284  7.313615640 -1.30391747       1 P4_H10.2.cover
## 285  7.313615640 -1.30391747       1 P4_H11.2.cover
## 286 -2.367515353 -6.56439923       1 P4_H12.2.cover
\end{verbatim}


\end{document}
