---
title: "vcf_clustering"
author: "EPermina"
date: "15 April 2018"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}


if(!require(tidyverse)){
  install.packages("tidyverse")
}

if(!require(SNPRelate)){ 
source("https://bioconductor.org/biocLite.R")
biocLite("SNPRelate")
}


if(!require(ISLR)) {
  install.packages("ISLR")
}

if(!require(cluster)) {
  install.packages("cluster")
}

if(!require(Rtsne)) {
  install.packages("Rtsne")
}

library(cluster) # for gower similarity and pam
library(Rtsne) # for t-SNE plot
library(tidyverse)
library(SNPRelate)
library(reshape2)
set.seed(1680) # for reproducibility
```

### Data manipulation, reading in

first_analysis/populations.snps.vcf.filtered.for.non-presence.presumed.0.samplesOnly.txt is a modified populations.snps.vcf file

1. all samples with no coverage of SNPs (./. only columns) were removed
1. vcf header has also been removed
1. ./. is replaced with 0/0:0:0,0

Finding relevant files on the users computer and reading these in
files:
populations.snps.vcf.Z.txt
populations.snps.vcf.Z.headless.sampleNames.trimmed.txt
populations.snps.vcf.split.info.genotype.coverage.txt

```{r, engine='bash', eval = FALSE}
cat populations.snps.vcf.Z.txt | sed 's/.single//g' | tail -n +15 | cut -f10- > populations.snps.vcf.Z.headless.sampleNames.trimmed.txt
```


```{r, engine='bash', eval = FALSE}

cat genotype.txt | sed 's#0/1#0.5#g' | sed 's#0/0#0#g' | sed 's#1/1#1#g' > genotype.01.txt
```




```{r, eval=FALSE}
# read in bash-splitted file
genotype_split<-read.delim("../populations.snps.vcf.split.info.genotype.coverage.txt", header = F, sep = '\t')
genotype_raw<-read.delim("../populations.snps.vcf.Z.headless.sampleNames.trimmed.txt", sep = "\t", header = TRUE) 

genotype_names_in_split <- paste(names, ".1.geno", sep="") # genotype names in split
coverage_names_in_split <- paste(names, ".2.cover", sep="") # coverage names in split
pro<- paste(names, ".3.pro", sep="") # coverage names in split

#bind 3 vectors
 z <- as.vector(rbind(genotype_names_in_split,coverage_names_in_split,pro)) 
names(genotype_split)<-z



```


Clustering - coverage and genotype
```{r}
genotype_split<-read.delim("genotype_split_object.tab")
coverage_cols<-grepl("2.cover", names(genotype_split))
genotype_cols<-grepl("1.geno", names(genotype_split))
coverage<-genotype_split[coverage_cols]
#get the names of the samples with > 0 presence

non_zero_coverage <- coverage[,colSums(coverage) > 0]
non_zero_names<-names(non_zero_coverage)



fileConn<-file("non_zero_names")
writeLines(non_zero_names, fileConn)
close(fileConn)

non_zero_names_parsed<-scan(file="non_zero_names.txt", what="character")

genotype<- read.delim("genotype.01.txt", sep = "\t", header = TRUE)
fileConn<-file("genotype_names")
writeLines(names(genotype), fileConn)
close(fileConn)

genotype_names_parsed<-scan(file="genotype_names.txt", what="character")
genotype_non_zero<-genotype[non_zero_names_parsed]

t.coverage<-t(as.matrix(coverage))
t.genotype<-t(as.matrix(genotype))
t.genotype_non_zero<-t(as.matrix(genotype_non_zero))

 # df3 = merge(t.coverage, t.genotype, by.x=row.names(t.coverage), by.y=row.names(t.genotype))

# Applying Ward Hierarchical Clustering
d_coverage   <- dist(t.coverage, method="euclidean")
ward_fit_coverage <- hclust(d_coverage, method="ward")
g4_ward_coverage <- cutree(ward_fit_coverage, k = c(4))
g8_ward_coverage <- cutree(ward_fit_coverage, k = c(8))

#using Gower via daisy
gower_dist_genotype <- daisy(t.genotype,
                    metric = "gower",
                    type = list(logratio = 3))
gower_mat_genotype <- as.matrix(gower_dist_genotype)

fit_genotype<-hclust(gower_dist_genotype, method = "median")

tsne_obj_genotype <- Rtsne(gower_dist_genotype, perplexity=15, is_distance = TRUE)
pam_fit_genotype <- pam(gower_dist_genotype, diss = TRUE, k = 3)
tsne_data_genotype <- tsne_obj_genotype$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit_genotype$clustering),
         name = row.names(t.genotype))


gower_dist_genotype_non_zero<-daisy(t.genotype_non_zero, metric = "gower", 
                                    type = list(logratio = 3))
gower_mat_genotype_non_zero<-as.matrix(gower_dist_genotype_non_zero)
fit_genotype_non_zero<-hclust(gower_dist_genotype_non_zero)

tsne_obj_genotype_non_zero <- Rtsne(gower_dist_genotype_non_zero, perplexity=15, is_distance = TRUE)
pam_fit_genotype_non_zero <- pam(gower_dist_genotype_non_zero, diss = TRUE, k = 3)
tsne_data_genotype_non_zero <- tsne_obj_genotype_non_zero$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit_genotype_non_zero$clustering),
         name = row.names(t.genotype_non_zero))



# Gower clusters for coverage:

gower_dist_coverage <- daisy(t.coverage,
                    metric = "gower",
                    type = list(logratio = 3))
gower_mat_coverage <- as.matrix(gower_dist_coverage)


tsne_obj_coverage <- Rtsne(gower_dist_coverage, perplexity=15, is_distance = TRUE)
pam_fit_coverage <- pam(gower_dist_coverage, diss = TRUE, k = 3)
tsne_data_coverage <- tsne_obj_coverage$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit_coverage$clustering),
         name = row.names(t.coverage))

```

Writing data out 

```{r}
write.table(tsne_data_genotype, file="genotype_clusters.csv", quote = FALSE, sep = ",")
write.table(g8_ward_coverage, file="coverage_groups.csv", quote = FALSE, sep = ",")
write.table(tsne_data_coverage, file = "coverage_clusters.csv", quote = FALSE, sep = ",")

```

Plots:
Ward clustered dendrogram for coverage:
```{r ward for coverage}

par(cex=0.4, mar=c(5, 8, 4, 1))
plot(ward_fit_coverage, xlab="coverage", ylab="", main="", sub="", axes=FALSE)
par(cex=1)
title(xlab="samples", ylab="Ward's D", main="coverage")
axis(2)

```

Genotype clusters:
```{r genotype clusters}
ggplot(aes(x = X, y = Y), data = tsne_data_genotype) +
  geom_point(aes(color = cluster))
```

```{r}
ggplot(aes(x = X, y = Y), data = tsne_data_genotype_non_zero) +
  geom_point(aes(color = cluster))
#shows we should try 5 groups instead
```


Dendrogram for genotypes:
```{r dendrogram for genotypes}
fit_genotype<-hclust(gower_dist_genotype, method = "ward.D")
par(cex=0.4, mar=c(5, 8, 4, 1))
plot(fit_genotype, xlab="genotype", ylab="", main="", sub="", axes=FALSE)
par(cex=1)
title(xlab="samples", ylab="ward", main="genotype")
axis(2)

```


```{r}
fit_genotype<-hclust(gower_dist_genotype_non_zero, method = "ward.D")
par(cex=0.4, mar=c(5, 8, 4, 1))
plot(fit_genotype_non_zero, xlab="", ylab="", main="", sub="", axes=FALSE)
par(cex=1)
title(xlab="samples", ylab="ward", main="genotype - non-zero coverage")
axis(2)
```


Samples in genotype clusters:

```{r}
tsne_data_genotype
```

Coverage clusters:

```{r coverage clusters - plot}
ggplot(aes(x = X, y = Y), data = tsne_data_coverage) +
  geom_point(aes(color = cluster))
```

Samples and coverage clusters:

```{r coverage in samples table}
tsne_data_coverage
```

Clustering SNPs
```{r SNP clusters}

m.coverage<-as.matrix(coverage)
m.genotype<-as.matrix(genotype)


 # df3 = merge(t.coverage, t.genotype, by.x=row.names(t.coverage), by.y=row.names(t.genotype))

# Applying Ward Hierarchical Clustering
d_coverage_snp   <- dist(m.coverage, method="euclidean")
ward_fit_coverage_snp <- hclust(d_coverage_snp, method="ward")
g4_ward_coverage_snp <- cutree(ward_fit_coverage_snp, k = c(4))
g8_ward_coverage_snp <- cutree(ward_fit_coverage_snp, k = c(8))

#using Gower via daisy
gower_dist_genotype_snp <- daisy(m.genotype,
                    metric = "gower",
                    type = list(logratio = 3))
gower_mat_genotype <- as.matrix(gower_dist_genotype)

fit_genotype<-hclust(gower_dist_genotype, method = "median")

tsne_obj_genotype <- Rtsne(gower_dist_genotype, is_distance = TRUE)
pam_fit_genotype <- pam(gower_dist_genotype, diss = TRUE, k = 3)
tsne_data_genotype <- tsne_obj_genotype$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit_genotype$clustering),
         name = row.names(t.genotype))


# Gower clusters for coverage:

gower_dist_coverage <- daisy(t.coverage,
                    metric = "gower",
                    type = list(logratio = 3))
gower_mat_coverage <- as.matrix(gower_dist_coverage)


tsne_obj_coverage <- Rtsne(gower_dist_coverage, is_distance = TRUE)
pam_fit_coverage <- pam(gower_dist_coverage, diss = TRUE, k = 3)
tsne_data_coverage <- tsne_obj_coverage$Y %>%
  data.frame() %>%
  setNames(c("X", "Y")) %>%
  mutate(cluster = factor(pam_fit_coverage$clustering),
         name = row.names(t.coverage))

```


